### ISSUE TITLE
Real time output handling

### ISSUE THREAD

**Issue by Taitava** [Taitava/obsidian-shellcommands#275]
### Discussed in https://github.com/Taitava/obsidian-shellcommands/discussions/64

So far, output has been handled after the whole shell command execution process has finished. That's perfectly fine for most shell commands, if they run quickly.

I will add a choice to start handling output before a shell command has finished executing. It will allow e.g. to get output text inserted to current note in multiple phases during a single shell command executon process.

The traditional way will also be still available, and it will stay as a default output handling mode.

#128 gave me some implementation ideas: https://github.com/Taitava/obsidian-shellcommands/pull/128/files?diff=unified&w=0.

TODO:
- [x] Add to changelog.
- [x] Documentation
- [x] Tests

**Comment by Taitava** [comment]
This feature is quite complete, I think. But I have one problem that I need to think about. If both `stdout` and `stderr` contain output, sometimes they will be outputted in a bit wrong order.

Example:
```powershell
echo Line0
[Console]::Error.WriteLine("Error line1")
[Console]::Error.WriteLine("Error line2")
echo Line1
echo Line2
echo Line3
echo Line4
echo Line5
echo Line6
echo Line7
echo Line8
echo Line9
```

Sometimes _Error line1_ and _Error line2_ are processed after _Line1_, or even later. According to my quick research, it looks like there's no way to guarantee the output order:
 - https://github.com/nodejs/node/issues/26516
 - https://stackoverflow.com/questions/57046930/node-js-spawn-keep-stdout-and-stderr-in-the-original-order

I have only tested this in PowerShell.

I'll continue thinking about this.

**Comment by Taitava** [comment]
> I have one problem that I need to think about. If both `stdout` and `stderr` contain output, sometimes they will be outputted in a bit wrong order.
> 
> ...
> 
> I'll continue thinking about this.

I have decided not to search for any solutions for this problem - at least at this time. This feature can be released even with this problem existing.

**Comment by Taitava** [comment]
Documentation is done. Still needs tests.

**Comment by Taitava** [comment]
Still need to make tests for the following output channels:
- [x] Notification balloon
- [ ] A separate test for Error balloon????
- [ ] Current file: caret position
- [ ] Current file: top
- [ ] Current file: bottom
- [ ] Open files (simply just open Welcome.md five times, first in a new window, then the rest in new tabs)
- [ ] Status bar
- [ ] Clipboard
- [ ] Ask after execution

**Comment by Taitava** [comment]
All tests are now created. The feature should be ready now, and will be released in `0.17.0`.

**Comment by Taitava** [comment]
Released now.
